### DISCUSSION TITLE
stm32: Missing LL header files ?

### DISCUSSION THREAD

**Discussion by erwango** [zephyrproject-rtos/zephyr#30369] [Announcements]
tl;dr: Please run `west update`

Following merge of [use generic LL headers](https://github.com/zephyrproject-rtos/zephyr/pull/28894) and [STM32 generic LL headers for further peripherals](#30163), STM32 drivers all make use of "generic LL headers". This changes were made as a solution to [Improve STM32 LL HAL usage](https://github.com/zephyrproject-rtos/zephyr/issues/28822).

As a consequence, LL headers are now provided by headers files available in `hal_stm32` module: https://github.com/zephyrproject-rtos/hal_stm32/tree/master/stm32cube/common_ll/include.

You might see building errors due a misalignment between Zephyr main repo version and the version of your `hal_stm32` module. Something like:
```
zephyrproject/zephyr/drivers/dac/dac_stm32.c:33:18: error: 'LL_DAC_CHANNEL_1' undeclared here (not in a function); did you mean 'LL_ADC_CHANNEL_1'?
33 | #define CHAN(n) LL_DAC_CHANNEL_##n
| ^~~~~~~~~~~~~~~
```
You can fix this issues using `west update`.

In order to avoid this kind of desync between zephyr main repo and modules, you can have a look to https://github.com/zephyrproject-rtos/zephyr/discussions/28651

Note: Same solution applies to missing `st/foo/stm32foobar-pinctrl.dtsi` files.
